magento/pwa-studio
# Moving npm-is into pwa-studio.

## Description

**npm-is** a utility package that validates if a given script is executed using the given package manager. For instance, `pwa-studio` can only be installed using `yarn` and if you try to install using `npm` it has to fail. @zetlen created this wonderful utility package a while ago which we have been using in the **preinstall** script to check validate if yarn is being used to install the package. Unfortunately in npm 7 update a critical piece of code that the package relies on has changed. Specifically the value of `process.env.npm_execpath` when it is invoked using `npx`. The script extensively relies on the value of `process.env.npm_execpath` to make the decision about what package manager is used to initiating the script.

Previously (npm 6 or less) if you initiated a script using `yarn` which internally uses `npx` the value of `process.env.npm_execpath` was `PATH_TO_THE_YARN_EXECUTABLE` but after the update, the value of `process.env.npm_execpath` has changed to `PATH_TO_THE_NPX_EXECUTABLE` instead. Due to this the validation has failed and `pwa-studio` installation fails with the following error:

![image](https://user-images.githubusercontent.com/35203638/113788528-bbb71a00-9702-11eb-8096-b6a8bf6a9943.png)

This PR addresses this issue by removing all references of `npm-is` and adding a note in the documentation to avoid using `npm`.

## Related Issue
Closes PWA-1660

## Acceptance
Should be able to validate if a script is initiated using a given package manager.

### Verification Stakeholders
@zetlen 
@dpatil-magento 

### Verification Steps
1. Checkout the branch.
2. Run `yarn install` and it should work fine.
3. Run `npm install` and it should fail.

## Checklist
* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
